src/api/docgen/render/markdown.ts

```diff
@@ -0,0 +1,193 @@
+import * as reflect from "jsii-reflect";
+
+/**
+ * Options for defining a markdown header.
+ */
+export interface MarkdownHeaderOptions {
+  /**
+   * Title to be displayed.
+   */
+  readonly title?: string;
+
+  /**
+   * Superscript.
+   *
+   * @default - No superscript
+   */
+  readonly sup?: string;
+
+  /**
+   * Preformat the header.
+   *
+   * @default false
+   */
+  readonly pre?: boolean;
+
+  /**
+   * Strikethough the title.
+   *
+   * @default false
+   */
+  readonly strike?: boolean;
+}
+
+/**
+ * Options for defining a markdown element.
+ */
+export interface MarkdownOptions {
+  /**
+   * Markdown header.
+   *
+   * @default - No header.
+   */
+  readonly header?: MarkdownHeaderOptions;
+
+  /**
+   * Id of the element.
+   *
+   * @default - The title will be used.
+   */
+  readonly id?: string;
+}
+
+/**
+ * Markdown element.
+ */
+export class Markdown {
+  /**
+   * An empty markdown element.
+   */
+  public static readonly EMPTY = new Markdown();
+
+  /**
+   * Sanitize markdown reserved characters from external input.
+   */
+  public static sanitize(line: string): string {
+    let sanitized = line.trim();
+
+    if (line.startsWith("-")) {
+      sanitized = sanitized.substring(1, line.length).trim();
+    }
+
+    return sanitized;
+  }
+
+  public static bold(text: string): string {
```

Comment:
Might be useful to add a custom formatted for underline which is encoded as `_underline_`

Reply:
What do you mean? To my understanding markdown doesn't have underlines. 

`_underline_` => _underline_

Reply:
I think it should be possible to add some custom markdown formatting which will render `_this_` with underline in the construct hub. In GitHub, it will render as *italics*.

Reply:
We can add later